weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -872,6 +967,49 @@ def table_query_stats(self, req: tsi.TableQueryStatsReq) -> tsi.TableQueryStatsR
 
         return tsi.TableQueryStatsRes(count=count)
 
+    def _table_delete(self, project_id: str, digest: str) -> None:
+        """
+        Internal function to delete a table and all of its rows.
+
+        1. Get the row_digests for the table
+        2. Delete the rows by row_digest
+        3. Delete the table by digest
+        """
+        # 1. get the row_digests for the table
+        query = """
+        SELECT row_digests
+        FROM tables
+        WHERE project_id = {project_id:String} AND digest = {digest:String}
+        """
+        parameters = {
+            "project_id": project_id,
+            "digest": digest,
+        }
+        query_result = self.ch_client.query(query, parameters=parameters)
+        row_digests = query_result.result_rows[0][0] if query_result.result_rows else []
+
+        # 2. delete the rows by row_digest
+        delete_query = """
+        DELETE FROM table_rows
+        WHERE project_id = {project_id:String} AND digest IN {row_digests:Array(String)}
+        """
+        parameters = {
+            "project_id": project_id,
+            "row_digests": row_digests,
+        }
+        self.ch_client.query(delete_query, parameters=parameters)
+
+        # 3. delete the table by digest
+        delete_table_query = """
```

Comment:
could also leave this, as a memory of the deleted table. tables with zero row_digests are considered deleted?